ruler.js

```diff
@@ -51,18 +76,54 @@ function createRuler() {
                 horizontalLine.style.left = `${x}px`;
                 horizontalLine.style.top = `${y + 13}px`;
         });
+
+        document.querySelectorAll("a").forEach((link) => {
+                link.addEventListener("mouseenter", enterLink);
+                link.addEventListener("mouseleave", leaveLink);
+        });
 }
 
 function removeRuler() {
         // Remove cursor and horizontal line
         const customCursor = document.getElementById("custom-cursor");
         const horizontalLine = document.getElementById("horizontal-line");
+        const parentRulerContainer = document.getElementById("parent-container");
+        const childRulerContainer = document.getElementById("child-container");
+
+        // Remove EventListeners
+        removeEventListener("mouseenter", enterLink);
```

Comment:
This should follow the same rule as the addEventListener. Should be removed from each link

Reply:
changes done